accession-commons-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/AccessioningServiceStepDefinitions.java

```diff
@@ -0,0 +1,143 @@
+
+    @Given("^already accessioned ([\\w-,]+)$")
+    @When("^user submits ([\\w-,]+) to accessioning service$")
+    public void submitObjectsToAccessioningService(String objects) {
+        tester.getOrCreate(toModels(objects));
+    }
+
+    @Then("^user received accessions: ([\\w-,]+)$")
+    public void iShouldReceiveAccessions(String objects) {
+        final String[] accessions = objects.split(",");
+        tester.getSingleVersionResults().assertAccessions(accessions);
+    }
+
+    @When("^user sends patch (\\w+) for accession ([\\w-]+)$")
+    public void userSendsPatchInputForAccessionAccession(String patchData, String accession) {
+        tester.patch(accession, TestModel.of(patchData));
+    }
+
+    @Then("^user should receive version list for ([\\w-]+) with (\\d+) version$")
+    public void userShouldReceivePatchForAccessionsWithVersion(String accession, int totalVersions) {
+        tester.getLastMultipleVersionResult().assertAccession(accession).assertTotalVersions(totalVersions);
+    }
+
+    @And("^patch must have versions increased$")
+    public void patchMustHaveVersionsIncreased() {
+        tester.getLastMultipleVersionResult().assertVersionAreIncreased();
+    }
+
+    @Then("^user receives accession does not exist$")
+    public void userReceivesAccessionDoesNotExistError() {
+        tester.getLastMethodResponse().assertThrow(AccessionDoesNotExistException.class);
+    }
+
+    @When("^user updates ([\\w-]+) patch (\\d+) with ([\\w-]+)$")
+    public void userUpdatesAccessionWithInput(String accession, int patch, String newData) {
+        tester.update(accession, patch, newData);
+    }
+
+
+    @And("^hash of version (\\d+) should be ([\\w-]+)$")
+    public void hashOfVersionShouldBeHash(int version, String hash) {
+        tester.getLastMultipleVersionResult().assertHash(version, hash);
+    }
+
+    @When("^user merges ([\\w-]+) with ([\\w-]+) reason: ([\\w ]+)$")
+    public void userMergesAWithB(String accessionA, String accessionB, String reason) {
+        tester.merge(accessionA, accessionB, reason);
+    }
+
+    @Then("^user receives a hash already exists exception$")
+    public void weShouldHaveAHashAlreadyExistsException() {
+        tester.getLastMethodResponse().assertThrow(HashAlreadyExistsException.class);
+    }
+
+    @Then("^user operation finished correctly$")
+    public void lastProcessFinishedOk() {
+        tester.getLastMethodResponse().assertNoException();
+    }
+
+    @Then("^user receives an accession already merged error$")
+    public void weShouldHaveAAccessionAlreadyMerged() {
+        tester.getLastMethodResponse().assertThrow(AccessionMergedException.class);
+    }
+
+    @When("^user retrieves objects: ([\\w-,]+)$")
+    public void userRetrievesObjectA(String values) {
+        tester.get(toModels(values));
+    }
+
+    private List<TestModel> toModels(String values) {
+        return Arrays.stream(values.split(",")).map(TestModel::of).collect(Collectors.toList());
+    }
```

Comment:
This method should be moved right after the `@Given` at the beginning of the class.